### PR TITLE
limit surface albedo between 0 and 1

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

This shouldn't be needed, but just to be safe.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/cache/surface_albedo.jl [hunk 1]
```diff
@@ -115,51 +115,61 @@ Calculate the direct surface albedo using the regression function of Jin et al.
...
-                    p[5] * σ +
-                    p[6] * σ * cosθ
-                ) * exp(
-                    p[7] +
-                    p[8] * cosθ +
-                    p[9] * cosθ^2 +
-                    p[10] * σ +
-                    p[11] * σ * cosθ,
+            if cosθ <= 0
+                return 0
```
Comment (Sbozzolo) [inline src/cache/surface_albedo.jl:119]:
```suggestion
                return zero(FT)
```

File: src/cache/surface_albedo.jl [hunk 2]
```diff
@@ -115,51 +115,61 @@ Calculate the direct surface albedo using the regression function of Jin et al.
...
+                        p[11] * σ * cosθ,
+                    )
+
+                # return the albedo (J11, eq. 1)
+                return min(
+                    1,
+                    max(
+                        0,
+                        rf(n, cosθ) - rf(n, cosθ) / rf(n0, cosθ) * f(cosθ, σ),
+                    ),
```
Comment (Sbozzolo) [inline src/cache/surface_albedo.jl:170]:
```suggestion
                return min(
                    one(FT),
                    max(
                        zero(FT),
                        rf(n, cosθ) - rf(n, cosθ) / rf(n0, cosθ) * f(cosθ, σ),
                    ),
```

File: src/cache/surface_albedo.jl [hunk 3]
```diff
@@ -177,25 +187,36 @@ function surface_albedo_diffuse(
...
-                q_clear[4] * n * σ
-
-            # cloudy sky (J11, eq. 5b)
-            q_cloud = α_model.q_cloud
-            α_cloud = q_cloud[1] + q_cloud[2] * n + q_cloud[3] * n * σ
-
-            # return the albedo
-            return could_fraction * α_cloud + (1 - could_fraction) * α_clear
+            if cosθ <= 0
+                return 0
```
Comment (Sbozzolo) [inline src/cache/surface_albedo.jl:191]:
```suggestion
                return zero(FT)
```

File: src/cache/surface_albedo.jl [hunk 4]
```diff
@@ -177,25 +187,36 @@ function surface_albedo_diffuse(
...
+                α_cloud = q_cloud[1] + q_cloud[2] * n + q_cloud[3] * n * σ
+
+                # return the albedo
+                return min(
+                    1,
+                    max(
+                        0,
+                        could_fraction * α_cloud +
+                        (1 - could_fraction) * α_clear,
+                    ),
```
Comment (Sbozzolo) [inline src/cache/surface_albedo.jl:217]:
```suggestion
                    one(FT),
                    max(
                        zero(FT),
                        could_fraction * α_cloud +
                        (1 - could_fraction) * α_clear,
                    ),
```

File: src/cache/surface_albedo.jl (hunk 1 above)
Comment (szy21) [reply src/cache/surface_albedo.jl:119]: Thanks! Is zero(FT) the same as FT(0)?

Comment (Sbozzolo) [reply src/cache/surface_albedo.jl:119]: No, zero(FT) is slightly better, probably because it is inferred and compiled

**Comment by szy21** [comment]
I also change the values in the test when cos(zenith_angle) is smaller or equal than 0.

**Pull request merged**